### ISSUE TITLE
Schema validation

### ISSUE THREAD

**Issue by romanpa** [everit-org/json-schema#38]
Hi

The schema below should catch an error in an object below.
But id does not :(

Map schemaMap = getSchemaMap();
 JSONObject rawSchema = new JSONObject(schemaMap);
 Schema schema = SchemaLoader.load(rawSchema);
schema.validate(....)

{
  "type" : "object",
  "id" : "urn:jsonschema:com:zzzzz:tests:commons:jsonschema:models:Person",
  "properties" : {
    "book" : {
      "type" : "object",
      "id" : "urn:jsonschema:com:zzzzz:tests:commons:jsonschema:models:PhoneBook",
      "properties" : {
        "numbers" : {
          "type" : "array",
          "items" : {
            "type" : "object",
            "id" : "urn:jsonschema:com:zzzzz:tests:commons:jsonschema:models:PhoneNumber",
            "properties" : {
              "number" : {
                "type" : "string",
                "pattern" : "\\+?\\d+"
              },
              "type" : {
                "type" : "string"
              }
            }
          }
        }
      }
    },
    "phoneBooks" : {
      "type" : "array",
      "items" : {
        "type" : "object",
        "$ref" : "urn:jsonschema:com:zzzzz:tests:commons:jsonschema:models:PhoneBook"
      }
    }
  }
}

{
  "book" : {
    "numbers" : [ {
      "number" : "22222"
    }, {
      "number" : "11111"
    } ]
  },
  "phoneBooks" : [ {
    "numbers" : [ {
      "number" : "22222"
    }, {
      "number" : "11111"
    }, {
      "number" : "aaaaa"
    } ]
  } ]
}

**Comment by erosb** [comment]
Please send a PR with a failing test as it is described in the [contributing guidelines](https://github.com/everit-org/json-schema/blob/master/CONTRIBUTING.md)

**Comment by romanpa** [comment]
Hi

I'am sorry but fork was not imported correctly into IntelliJ and i am very short on time, so i created a gist

https://gist.github.com/romanpa/0ee3437f59f634821d8c03af2eaf69d3

Regards,
romanpa

**Comment by erosb** [comment]
I've spent quite some time on examining this issue, and I found the followings:

 * the `"$ref"` references are not resolved during schema loading if the `"type"` property is present. So a definition like `{ "type": "object", "$ref": "..." }` will have the same effect as just `{ "type": "object" }`. This is clearly a bug to be fixed, probably in the next release.
 * on the other hand, supporting URNs won't happen any time soon. Existing JSON Schema implementations also seem to vary in supporting it.
